kubetest/main.go

```diff
@@ -295,10 +299,28 @@ func complete(o *options) error {
 		<-interrupt.C // Drain value
 	}
 
+	ctx := context.Background()
+	var cancel context.CancelFunc
+
 	if timeout > 0 {
 		log.Printf("Limiting testing to %s", timeout)
 		interrupt.Reset(timeout)
+		ctx, cancel = context.WithTimeout(ctx, timeout)
+	} else {
+		ctx, cancel = context.WithCancel(ctx)
+	}
+
+	{
+		// Exit the kubetest context on a signal
```

Comment:
It seems like this only does anything if we happen to be inside get nodes?

Reply:
Correct.  But if/when we merge the grab-logs-via-SSH code, it'll exit that also.  You'll see that in that PR we can exit the `Dial` function easily, but that the SSH library doesn't have built-in support for it (yet).

Using this doesn't stop the existing mechanisms from working, though I do think we want to move them over.

I split this out so this change wouldn't sneak-in in the bigger PR.  And TBH use of the context is still a little bit unclear to me, other than that we should be using it as of Go 1.7, and so I thought maybe we could rope in a go-guru to peek at the second commit!